Issue #92 - New /flag command.

PureKrome opened the pull request:
- Added new /flag command.
- Added unit tests to test the new command.
- Added DB migration for a new DB field.

PureKrome committed:
- Added new /flag command.
- Added unit tests to test the new command.
- Added DB migration for a new DB field.

davidfowl commented on JabbR/Services/ChatService.cs:
```diff
@@ -490,5 +745,24 @@ internal static void ValidateNote(string note)
+
+        internal static string GetCountry(string isoCode)
+        {
+            return String.IsNullOrWhiteSpace(isoCode) ||
```
Too much stuff in here, doesn't need to be this complicated. 

```
if(String.IsNullOrEmpty(isoCode)) {
    return null;
}

string country;
if(Countries.TryGetValue(isoCode, out country)) 
{
       return country;
}

return null;
```

PureKrome replied:
Also NP :)

davidfowl commented on JabbR/Commands/CommandManager.cs:
```diff
@@ -809,5 +815,26 @@ private void HandleAfk(ChatUser user, string[] parts)
+
+        private void HandleFlag(ChatUser user, string[] parts)
+        {
+            string isoCode = String.Join(" ", parts.Skip(1)).Trim().ToLowerInvariant();
```
Don't do this. Just take parts[1] and do ToLowerInvariant

PureKrome replied:
NP :)

PureKrome committed:
- Refactored some complex /flag code to make it more simple, based on feedback from @DFowler.
- Refactored new-user-template splitting user details separate from the gravatar.
- Tweaked .user, .note and .flag css to handle new template formatting.
